src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -33,6 +35,85 @@ public override SarifLogVersionOne VisitSarifLogVersionOne(SarifLogVersionOne v1
             return null;
         }
 
+        internal CodeFlow CreateCodeFlow(CodeFlowVersionOne v1CodeFlow)
+        {
+            CodeFlow codeFlow = null;
+
+            if (v1CodeFlow != null)
+            {
+                codeFlow = new CodeFlow
+                {
+                    Message = CreateMessage(v1CodeFlow.Message),
+                    Properties = v1CodeFlow.Properties
+                };
+
+                if (v1CodeFlow.Locations != null && v1CodeFlow.Locations.Count > 0)
+                {
+                    _codeFlowLocationNestingLevel = 0;
+
+                    if (v1CodeFlow.Locations[0].Step == 0)
+                    {
+                        // If the steps are zero-based, add 1 to comply with the v2 spec
+                        _codeFlowLocationStepAdjustment = 1;
```

Comment:
Nice! 😄 

But (I hate to be a killjoy 😄 ): Bug: This means that `step` will not round-trip. If it was 0-based on the way in from v1 to v2, you'll have forgotten about that when you convert back to v1. You need to put the v1 `step` in the property  bag.

... as usual, unless and until we get tired of worrying about this stuff and stick the whole v1 run in the v2 run's property bag.